lib/netext/httpext/transport.go

```diff
@@ -115,9 +115,20 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["status"] = "0"
 		}
 	} else {
-		if enabledTags.Has(stats.TagURL) {
-			u := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}
-			tags["url"] = u.Clean()
+		urlEnabled := enabledTags.Has(stats.TagURL)
+		nameEnabled := enabledTags.Has(stats.TagName)
+		if urlEnabled || nameEnabled {
+			cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
+			if urlEnabled {
+				tags["url"] = cleanURL
+			}
+			if _, ok := tags["name"]; !ok && nameEnabled {
```

Comment:
I am pretty sure I am becoming a PITA, but can you move the `_, ok := tags["name"]; !ok` in the above if? or better yet in the `nameEnabled` ;)

Reply:
> I am pretty sure I am becoming a PITA

Only a bit :)

Is 413d0fe0 what you had in mind? I think I preferred the previous version, but out of curiosity, why is this better?

Reply:
you could have had `name` set and `url` disabled and you would've stilled cleaned the url and then found out you didn't need it